File: examples/breadcrumbs2/breadcrumbs.rb

breadcrumbs = [
  { :title=>'TOP',          :path=>'/' },
  { :title=>'Kwartz',       :path=>'/kwartz' },
  { :title=>'Examples',     :path=>'/kwartz/examples' },
  { :title=>'breadcrumbs2', :path=>nil },
]
require 'stringio'

_out = StringIO.new; _out << '<html>
  <body>

';     last_index = breadcrumbs.length - 1 
     breadcrumbs.each_with_index do |item, i| 
       if i < last_index 
 _out << '        <a href="'; _out << ( item[:path] ).to_s; _out << '">'; _out << ( item[:title] ).to_s; _out << '</a> &lt;
';       else 
 _out << ( item[:title] ).to_s;       end ; _out << '
';     end 
 _out << '
  </body>
</html>
';
print _out.string

File: examples/breadcrumbs/.expected/breadcrumbs.php
<html>
  <body>

    <div id="breadcrumbs">
<?php foreach ($breadcrumbs as $crumb) { ?>
      <a href="<?php echo $crumb['path']; ?>"><?php echo $crumb['name']; ?></a>
      &gt;
<?php } ?>
      <?php echo $title; ?>
    </div>

  </body>
</html>
